wg-meetings/2018-10-17.md

```diff
@@ -0,0 +1,72 @@
+
+## Attendees
+  -  Mike Kaufman
+  -  Andreas Madsen: @andreasmadsen
+  -  Thomas Watson @watson
+  -  Matheus Marchini @mmarchini
+  -  Daniel Beckert @Drieger
+  -  Michael Dawson @mhdawson
+  -  Stephen Belanger @qard
+  -  Ali Sheikh @ofrobots
+  -  Kelvin Jin @kjin
+  -  Natalie Wolf
+  -  Bryan Clement
+
+## Links
+- **This Meeting**:
+  - Minutes:  https://github.com/nodejs/diagnostics/blob/master/wg-meetings/2018-10-17.md
+  - Notes:    https://docs.google.com/document/d/1l-hbheqECIKO0TMNymMweNX3LvtMxA2Id3dY7qUAoAQ/edit
+  - You Tube: http://youtu.be/3linTf2Z1N8
+- **Previous Meeting**: 
+  - Minutes:  https://github.com/nodejs/diagnostics/blob/master/wg-meetings/2018-10-03.md
+  - Notes:    https://docs.google.com/document/d/1zPXlGg3-yx5fthXOIw9l66fbFKt6edE_aK7infiyo4I/edit
+  - You Tube: http://youtu.be/VzBdlYATQfE
+
+## Agenda
+
+Extracted from **diag-agenda**  labelled issues and pull requests from the **nodejs org** prior to the meeting.
+
+### nodejs/diagnostics
+
+  - Session at Collab Summit Vancouver 2018 [#229](https://github.com/nodejs/diagnostics/issues/168)
+    - Intro to key initiatives to get broader awareness/interest
+    - Async hooks formalization, points at wanting a different API than Async hooks, need to work
+    on next steps.
+    - [#245](https://github.com/nodejs/diagnostics/issues/245) summarizes next steps for
+    port mortem/promises discussion
+
+  - Diagnostics "Best Practices" Guide? - [#211](https://github.com/nodejs/diagnostics/issues/211)
+    - Mike from NodeSource has written some content
+    - Need to find way to move this forward.
+    - Gireesh will setup separate meeting to get this moving forward
+
+  - Diag WG Deep Dives - topics [#168](https://github.com/nodejs/diagnostics/issues/168)
+     - @mike-kaufman will try to schedule a deep-dive on async context for sometime in November
+
+  - Expectation about tier of support from diagnostic tools and VMs [#157](https://github.com/nodejs/diagnostics/issues/157)
+    - nothing new to report
+
+  - \[async_hooks\] stable API - tracking issue [#124](https://github.com/nodejs/diagnostics/issues/124)
+    - no update this week
+    - Open PR [21313](github.com/nodejs/node/pull/21313), eliminates need for destroy callback & significantly improves perf
+         - Some discussion of reviving async resource.  Goal is to
+           remove the destroy hook, or so optional that it does not interfere with performance, PR
+           to remove promise from Promise wrap.  Will make escape analysis easier but requires
+          work on the V8 side to use escape analysis in emitting the hook
+    - Ali need Yang and Matteo to go into more details. Ali will schedule a separate time
+    for the discussion.
+  
+  - Async-context formalization and diagnostics support [#107](https://github.com/nodejs/diagnostics/issues/107)
+    - will schedule a deep-dive for november
+    - Natalie & Bryan from new relic are going to do some code
```

Comment:
Is this in a public repo? I would like to help :)

Reply:
@mmarchini  - We'd love any help you  can provide here.  A few places to dig in:
   -  There's a number of //todos now listed in #107. 
   -  @NatalieWolf started a Node.JS implementation:  https://github.com/NatalieWolfe/node-context-continuation.   I think this could benefit from playing with it, as well as filling in the details around the ready context hook-up.
  - I'm really interested in what this would look like when wired up at the VM level, so I think there's opportunity to do this as a strawman, in V8 and/or Chakra-Core.  If Chakra-Core, I can help with guidance on what needs to be written (I haven't found the time...).

Reply:
Tagging the correct Natalie Wolf so they can see this :) @nataliewolfe